src/codec/builtin/BigDecimalCodec.ts

```diff
@@ -3,44 +3,52 @@ import {CodecUtil} from './CodecUtil';
 import {FixSizedTypesCodec} from './FixSizedTypesCodec';
 import {BitsUtil} from '../../util/BitsUtil';
 
-function bytes2hexString(uint8array: Uint8Array): string { // buffer is an ArrayBuffer
-    return '0x' + [...uint8array].map(x => x.toString(16).padStart(2, '0')).join('');
-}
-
 /** @internal */
 export class BigDecimalCodec {
 
     static decode(clientMessage: ClientMessage): string {
         const buffer = clientMessage.nextFrame().content;
         const contentSize = FixSizedTypesCodec.decodeInt(buffer, 0);
-        const body = new Uint8Array(buffer.slice(BitsUtil.INT_SIZE_IN_BYTES, BitsUtil.INT_SIZE_IN_BYTES + contentSize));
-        const scale = FixSizedTypesCodec.decodeInt(buffer, BitsUtil.INT_SIZE_IN_BYTES + contentSize);
+        let body = buffer.slice(BitsUtil.INT_SIZE_IN_BYTES, BitsUtil.INT_SIZE_IN_BYTES + contentSize);
+        const byteArray = new Uint8Array(body);
```

Comment:
Why are we still wrapping it into Uint8Array?

Reply:
In order to be able to negate bytes we need a ArrayBuffer view. UInt8Array works for this case.

Reply:
I get what you are trying to do, but how this is different than what you are doing?
```js
const buf = Buffer.from([1,2,3,4,5]);
const negated = Buffer.allocUnsafe(buf.length);

for (let i = 0; i < buf.length; i++) {
    negated[i] = ~buf[i];
}
```

Reply:
thanks I did not know I can do that

Reply:
Now it is much simpler